src/plugin/src/main/java/com/defenseunicorns/uds/keycloak/plugin/clientpolicy/executor/UDSClientPolicyPermissionsExecutor.java

```diff
@@ -0,0 +1,104 @@
+                    case UPDATE:
+                        logger.debug("Updating existing Client with Client ID: {}", clientCRUDContext.getTargetClient().getClientId());
+                        if (!isOwnedByUDSOperator(clientCRUDContext.getTargetClient()) && !hasBackwardsCompatibleClientName(clientCRUDContext.getTargetClient())) {
+                            throw new ClientPolicyException(Errors.UNAUTHORIZED_CLIENT, "The Client doesn't have the " + ATTRIBUTE_UDS_OPERATOR + "=" + ATTRIBUTE_UDS_OPERATOR_VALUE + " attribute. Rejecting request.");
+                        }
+                        setUDSOperatorAsOwner(clientCRUDContext.getProposedClientRepresentation());
+                        break;
+
+                    case REGISTER:
+                        logger.debug("Creating new Client with Client ID: {}", clientCRUDContext.getProposedClientRepresentation().getClientId());
+                        setUDSOperatorAsOwner(clientCRUDContext.getProposedClientRepresentation());
+                        break;
+
+                    case VIEW:
+                    case UNREGISTER:
+                        logger.debug("Viewing or deleting Client with Client ID: {}", clientCRUDContext.getTargetClient().getClientId());
+                        if (!isOwnedByUDSOperator(clientCRUDContext.getTargetClient()) && !hasBackwardsCompatibleClientName(clientCRUDContext.getTargetClient())) {
+                            throw new ClientPolicyException(Errors.UNAUTHORIZED_CLIENT, "The Client doesn't have the " + ATTRIBUTE_UDS_OPERATOR + "=" + ATTRIBUTE_UDS_OPERATOR_VALUE + " attribute. Rejecting request.");
+                        }
+                        break;
+                }
+            }
+        }
+    }
+
+    String getAuthenticatedClientId(ClientCRUDContext clientCRUDContext) {
+        if (clientCRUDContext.getAuthenticatedClient() == null) {
+            return null;
+        }
+        return clientCRUDContext.getAuthenticatedClient().getClientId();
+    }
+
+    boolean isOwnedByUDSOperator(ClientModel client) {
+        return ATTRIBUTE_UDS_OPERATOR_VALUE.equals(client.getAttribute(ATTRIBUTE_UDS_OPERATOR));
+    }
+
+    boolean hasBackwardsCompatibleClientName(ClientModel client) {
+        return client.getClientId().startsWith("uds");
+    }
+
+    @Override
+    public void setupConfiguration(ClientPolicyExecutorConfigurationRepresentation config) {
+    }
+
+    @Override
+    public Class<ClientPolicyExecutorConfigurationRepresentation> getExecutorConfigurationClass() {
+        return ClientPolicyExecutorConfigurationRepresentation.class;
+    }
+
+    @Override
+    public String getProviderId() {
+        return FullScopeDisabledExecutorFactory.PROVIDER_ID;
+    }
+
+    private void setUDSOperatorAsOwner(ClientRepresentation rep) {
+        if (rep.getAttributes() == null)
+            rep.setAttributes(new java.util.HashMap<>());
+        rep.getAttributes().put(ATTRIBUTE_UDS_OPERATOR, ATTRIBUTE_UDS_OPERATOR_VALUE);
```

Comment:
We will probably need to mirror the Client Registration Policies here (potentially without a list of trusted hosts as this is already covered by the Authentication and Istio).

Reply:
I think we probably would want to keep the restrictions but I might need to further review. If there are `Package` CRs that could be created, but fail reconciliation because of this policy - we may want to review our validator code and block `Package` CRs if the SSO config would cause an error (this would surface the failure quicker).

Reply:
Ok, makes sense. I've implemented missing checks and enforcement.